components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -222,13 +215,28 @@ private fun load(
             loaders.forEach { loader ->
                 val result = loader.load(context, request, resource)
 
-                if (result != IconLoader.Result.NoResult) {
-                    return Pair(result, resource)
+                val icon = decodeIcon(result, decoders, desiredSize)
+
+                if (icon != null) {
+                    return icon to resource
```

Comment:
nit: In this context I think `Pair(x ,y)` is much more readable. Using `to` makes much more sense inside something like `mapOf()`.